pdc/apps/tree/views.py

```diff
@@ -0,0 +1,302 @@
+#
+# Copyright (c) 2015 Red Hat
+# Licensed under The MIT License (MIT)
+# http://opensource.org/licenses/MIT
+#
+
+from pdc.apps.common import viewsets
+from .models import (Tree, TreeVariant,)
+from .serializers import (TreeSerializer, TreeVariantSerializer)
+from .filters import (TreeFilter, TreeVariantFilter)
+
+
+class TreeViewSet(viewsets.PDCModelViewSet):
+    """
+    ##Overview##
+
+    This page shows the usage of the **Tree API**, please see the
+    following for more details.
+
+    ##Test tools##
+
+    You can use ``curl`` in terminal, with -X _method_ (GET|POST|PUT|DELETE),
+    -d _data_ (a json string). or GUI plugins for
+    browsers, such as ``RESTClient``, ``RESTConsole``.
+
+    ## Customize Output ##
+
+    There are two query params that you can use to customize your output.
+
+    `fields`:          string, can be set multiple times, to demand what fields you want to include;
+
+    `exclude_fields`:  string, can be set multiple times, to demand what fields you do NOT want.
+
+    __NOTE__: If both given, `exclude_fields` *rules* `fields`.
+
+    """
+    model = Tree
+    queryset = Tree.objects.all().order_by('tree_id')
+    serializer_class = TreeSerializer
+    filter_class = TreeFilter
+
+    def list(self, request, *args, **kwargs):
+        """
+        __Method__:
+        GET
+
+        __URL__: $LINK:tree-list$
+
+        __Query Params__:
+
+        %(FILTERS)s
+
+        __Response__:
+
+            # paged lists
+            {
+                "count": int,
+                "next": url,
+                "previous": url,
+                "results": [
+                    {
+                        "tree_id": string,
+                        "tree_date": date,
+                        "variant": string
+                        "arch": string,
+                        "deleted": bool,
+                        "content": json_data,
+                        "content_format": [string, ...],
+                        'url':                    string,
+                    },
+                    ...
+            }
+        """
+        return super(TreeViewSet, self).list(request, *args, **kwargs)
```

Comment:
I've checked out where this is done in PDC upstream as well and it's from the initial import, so there's no real reasoning to be found there either—I feel something cargo-cultish happening here :P.

I can only imagine that this is to get automatic API docs, tailored in wording, i.e. not simply copied from the parent class. Hmm. If so, it still rubs me the wrong way to manually call the super method(s) instead of letting Python do the job—I know we're not trying to win any performance contests (yet), but it also clutters up tracebacks.